src/core/ucc_lib.c

```diff
@@ -18,3 +28,70 @@ ucc_status_t ucc_init_version(unsigned api_major_version,
 {
     return UCC_ERR_NOT_IMPLEMENTED;
 }
+
+ucc_status_t ucc_lib_config_read(const char *env_prefix, const char *filename,
+                                 ucc_lib_config_t **config_p)
+{
+    ucc_lib_config_t *config;
+    ucc_status_t      status;
+    int               full_prefix_len;
+    const char       *base_prefix = "UCC_";
+
+    config = ucc_malloc(sizeof(*config), "lib_config");
+    if (config == NULL) {
+        ucc_error("failed to allocated %d bytes for lib_config",
+                  sizeof(*config));
+        status = UCC_ERR_NO_MEMORY;
+        goto err;
+    }
+    /* full_prefix for a UCC lib config is either just
+          (i)  "UCC_" - if not "env_prefix" is provided, or
+          (ii) "UCC_AAA_" - where AAA is the value of "env_prefix".
+       Need to allocate enough space to build the proper prefix str: 2 more
+       chars for "_" and "\0". */
+    full_prefix_len =
+        strlen(base_prefix) + (env_prefix ? strlen(env_prefix) : 0) + 2;
```

Comment:
is it should be +1 instead of +2 at the end if env_prefix is null?

Reply:
yes, but do you want process that case separately? i thought 1 byte does not worth extra hassle.

Reply:
sure, let's leave it as is